platform/pulp/platform/models/repository.py

```diff
@@ -1,10 +1,268 @@
 """
-Django models related to repositories.
+Repository related Django models.
 """
+from datetime import datetime
 
-from pulp.platform.models import Model
+from django.db import models
+from django.contrib.contenttypes import fields
 
+from pulp.platform.models import Model, Notes, Scratchpad, Config
 
-class Distributor(Model):
+
+class Content(Model):
     # placeholder
-    pass
+    type = models.TextField()
+    created = models.DateTimeField(auto_now_add=True)
+
+
+class Repository(Model):
+    """
+    Collection of content.
+
+    Fields:
+
+    :cvar name: The repository name.
+    :type name: models.TextField
+
+    :cvar: description: An optional description.
+    :type: models.TextField
+
+    :cvar type: The repository type.
+    :type type: models.TextField
+
+    :cvar last_content_added: When content was last added.
+    :type last_content_added: models.DateTimeField
+
+    :cvar last_content_removed: When content was last removed.
+    :type last_content_removed: models.DateTimeField
+
+    Relations:
+
+    :cvar scratchpad: Arbitrary information stashed on the repository.
+    :type scratchpad: fields.GenericRelation
+
+    :cvar notes: Arbitrary repository properties.
+    :type notes: fields.GenericRelation
+
+    :cvar content: Associated content.
+    :type content: models.ManyToManyField
+    """
+    name = models.TextField(db_index=True, unique=True)
```

Comment:
It's my understanding that the need for validation was imposed by mongod key restrictions.  Since the *name* is part of the natural key, I wonder if we should restrict characters that impact the REST API.  Like things that are not URL friendly.